velox/exec/GroupingSet.cpp

```diff
@@ -277,13 +277,14 @@ void GroupingSet::initializeGlobalAggregation() {
 
   // Row layout is:
   //  - null flags - one bit per aggregate,
-  // - uint32_t row size,
+  //  - uint32_t row size,
   //  - fixed-width accumulators - one per aggregate
   //
   // Here we always make space for a row size since we only have one
   // row and no RowContainer.
   int32_t rowSizeOffset = bits::nbytes(aggregates_.size());
-  int32_t offset = rowSizeOffset + sizeof(int32_t);
+  // Fixed-width accumulators must be word aligned.
```

Comment:
Where does this requirement come from and why do we round to 16 and not 8?

Reply:
Without this, the LongDecimal accumulator is segfaulting as the memory address is not aligned. Hence, rounding off to 2 words, the size of LongDecimal.

The requirement comes from the implementation of 
```
void extractAccumulators(char** groups, int32_t numGroups, VectorPtr* result)
      override {
    BaseAggregate::template doExtractValues<TAccumulator>(
        groups, numGroups, result, [&](char* group) {
          return *BaseAggregate::Aggregate::template value<TAccumulator>(group);
        });
  }
```
where 
```
T* value(char* group) const {
    return reinterpret_cast<T*>(group + offset_);
  }
```

Reply:
Thank you for explaining. It appears to me that this code / requirement is highly-specific to accumulators of type LongDecimal which require 16-byte alignment. At the same time the comment sounds generic and suggests that any fixed-width accumulator must be work aligned. Here, we have only one type of accumulators requiring two word alignment. I wonder if there is a way to generalize this code. If not, let's clarify in the comment that we require 16-byte (2-word) alignment to support accumulator types that use __uint128_t.

Reply:
I can make this comment. But in general, I thought unaligned access is not portable across architectures. So it should apply to all types.
We can tighten this if you would like by checking all the types in the vector of accumulators and aligning by the largest type.

Reply:
> in general, I thought unaligned access is not portable across architectures

This is so, but the comment says "word aligned", while the code makes it "two-word aligned". 

> We can tighten this ... by checking all the types in the vector of accumulators and aligning by the largest type.

This might be more robust and less specific to int128_t used by LongDecimal accumulator.